### ISSUE TITLE
SpEL expression in @VaultPropertySource annotation not expanded with enabled renewal

### ISSUE THREAD

**Issue by erikrz** [spring-projects/spring-vault#420]
I am using the annotation with SpEL, as the path inside Vault changes as a requirement.

```
@VaultPropertySource(value="kv/#{'${MY_ENVIRONMENT}'}/domain/apidomain", renewal = VaultPropertySource.Renewal.OFF)
```

I see the VaultPropertySource listed correctly in the Spring Environment propertySourceList, the value of my System Environment variable is replaced and I can access the values.

But if I use 

```
@VaultPropertySource(value="kv/#{'${MY_ENVIRONMENT}'}/domain/apidomain", renewal = VaultPropertySource.Renewal.ROTATE)
```

or 

```
@VaultPropertySource(value="kv/#{'${MY_ENVIRONMENT}'}/domain/apidomain", renewal = VaultPropertySource.Renewal.RENEW)
```

The `LeaseAwareVaultPropertySource` is also listed in the Spring Environment propertySourceList, but my SpEL expression is not being replaced and can't access the values, due to the request being made without replacing the SpEL expression.

I am using v2 engine

**Comment by mp911de** [comment]
Thanks for report. SpEL expansion works for non-renewable property sources sort of by accident. We never explicitly added property expansion but rather register beans through `BeanDefinitionBuilder` and use the property source path as `String`.

For renewable/rotating secrets, we create upfront a `RequestedSecret` and so we bypass the container's built-in SpEL expansion. We need to consistently call `Environment.resolveRequiredPlaceholders(…)` on bean registration to resolve property placeholders eagerly.

Flagging this ticket as an enhancement as we should also document that feature.

**Comment by mp911de** [comment]
That's implemented (fixed) and backported to 2.1.x now. Care to give `2.1.3.BUILD-SNAPSHOT` a spin?

**Comment by erikrz** [comment]
It works wonders. Thank you!
